pkg/monitor/monitor.go

```diff
@@ -62,80 +67,43 @@ func (m *Monitor) Run(ctx context.Context) (retErr error) {
 		}
 	}
 
-	var sd *comp_v1.ServiceDescriptor
-	sd, err = buildServiceDescriptor(m.ServiceDescriptorName, m.Location, m.Version)
-	if err != nil {
-		return err
-	}
-
-	err = m.createServiceDescriptor(ctx, sd)
+	err = m.createServiceDescriptor(ctx, m.ServiceDescriptor)
 	if err != nil {
 		return err
 	}
 
 	m.Logger.Info("Initialized ServiceDescriptor for synthetic checks")
 
-	// Wait until the corresponding ServiceInstance is created.
-	// Sleeps for a short period of time, which for now should be enough I think
-	time.Sleep(m.ExpectedProcessingTime)
-
-	err = m.verifyUpsServiceInstance(ctx, m.ServiceDescriptorName, m.Version)
+	err = wait.PollImmediate(pollDelay, m.ExpectedProcessingTime, m.verifyServiceDescriptorStatus(m.ServiceDescriptorName))
```

Comment:
We can avoid polling to get immediate results using `UntilWithSync()` from `"k8s.io/client-go/tools/watch"`. See how it is used in Smith's integration tests https://github.com/atlassian/smith/blob/e367e27c0d09db8c88715fd76122be086bacd15a/it/utils_for_tests.go#L89-L99